app/src/main/scala/org/alephium/explorer/persistence/queries/OutputQueries.scala

```diff
@@ -27,11 +26,7 @@ object OutputQueries {
 
   /** Inserts outputs or ignore rows with primary key conflict */
   def insertOutputs(outputs: Iterable[OutputEntity]): DBActionW[Int] =
-    if (outputs.isEmpty) {
-      DBIOAction.successful(0)
-    } else {
-      val placeholder = paramPlaceholder(rows = outputs.size, columns = 10)
-
+    QueryUtil.splitUpdates(rows = outputs, queryRowParams = 10) { (outputs, placeholder) =>
```

Comment:
that's the number  of columns in the `outputs` and `inputs` table

Reply:
I see. So the maximal batch is 128?

Reply:
The limit is `Short.MaxValue = 32767` of `?`. We can't get more bind variables for the prepred statement. So if a table has 10 columns, we can batch around 3270 entries

Reply:
Thanks.That sounds too many entries, and it will create a lot of prepare statement cache. Maybe we could bound it to `100` or `200`?

Reply:
we had this discussion and we said we  will first see how it performs and see if we need to bound later.
Knowing that we will insert a lot of entries only when syncing from scratch. Once we are up-to date with the node, we only insert very few blocks at a time as we sync every 5 sec.

Reply:
the issue of too many prepared statement in cache might need a long time to produce. Therefore it's safer to limit to a lower number by default.

Reply:
## Benchmarks to supplement this discussion
Benchmarks for writing input entity with limits of `100`, `200` & `Short.MaxValue`. 

![image](https://user-images.githubusercontent.com/1773953/158319981-9382eccf-e817-42aa-826a-f72029012fdf.png)

`Short.MaxValue` performs better. The bigger the batch size, the better it performs. 

## Cost of cache
 3 million `INSERT INPUT …` queries cached in a `ConcurrentHashMap[String, _]` cost roughly
- 70.8 mb in `java.lang.String` instances
- 16.8 mb in `ConcurrentHashMap#Node` instances

Our machine has 8GB RAM and cache is barely using 100MB for 3 million queries. Even if it was using 1GB, we'd still have a lot of room.

## Thoughts
Saving CPU time and Connection pool (release connections quick to serve more queries) time over RAM seems more important. We have RAM but we don’t have as many cores. We know slow query is problem which is fixed using the max limit `Short.MaxValue` (the benchmarks). But we don’t know if cache is a problem yet. Changing it to 100 or 200 we would be solving a problem that does not exist yet. Like you said ".. might need a long time to produce ...", **might not** be a problem yet. 

## Changing the limit 
If we want to use 100 or 200 regardless, the limit can be easily changed by setting the value [here - `val postgresDefaultParameterLimit: Short = Short.MaxValue`](https://github.com/alephium/explorer-backend/blob/master/app/src/main/scala/org/alephium/explorer/persistence/queries/QueryUtil.scala#L33).

Please let me know how you want this to proceed.

Reply:
@simerplaha that's great benchmark. Did you benchmark with 200 `?`, or 200 rows?

Reply:
With 200 `?`

Reply:
My guess is that the performance of inserting with `200 rows` per query should be similar to the performance of using `Short.MaxValue`

Reply:
@simerplaha Let's say the limit of parameters (i.e. `?`) is N, and each query has `M` parameters, then each insertion can insert upto `N/M` rows. If `N = 200` and `M=10`, then each batch can insert at most 20 rows. This would be very slow. However, if test with `N = 2000`, then each batch would insert 200 rows. Let me know if I am clear enough now.

Reply:
You are right. Performance is similar and caching should improve due to reduced randomness in query size.  

![image](https://user-images.githubusercontent.com/1773953/158512452-2fdbcecc-f41f-4e5b-bae2-e84beff21227.png)

![image](https://user-images.githubusercontent.com/1773953/158512485-13b4ae46-3d30-4a7b-923b-98c55921d10e.png)

I will submit a PR.